tests/v3_api/common.py

```diff
@@ -836,3 +842,26 @@ def wait_until_available(client, obj, timeout=DEFAULT_TIMEOUT):
             msg = 'Timeout waiting for [{}:{}] for condition after {}' \
                   ' seconds'.format(obj.type, obj.id, delta)
             raise Exception(msg)
+
+
+def delete_node(aws_nodes):
+    for node in aws_nodes:
+        AmazonWebServices().delete_node(node)
+
+
+def cluster_cleanup(client, cluster, aws_nodes=None):
+    if RANCHER_CLEANUP_CLUSTER:
+        delete_cluster(client, cluster)
+        if aws_nodes is not None:
+            delete_node(aws_nodes)
+    else:
+        env_details = "env.CATTLE_TEST_URL='" + CATTLE_TEST_URL + "'\n"
+        env_details += "env.ADMIN_TOKEN='" + ADMIN_TOKEN + "'\n"
+        env_details += "env.CLUSTER_NAME='" + cluster.name + "'\n"
+        create_config_file(env_details)
+
+
+def create_config_file(env_details):
+        file = open(env_file, "w")
```

Comment:
`file = open(env_file, "w")` will overwrite the existing file. Since we may be adding multiple cluster names, we should use "a" to append. We should also move writing `env.CATTLE_TEST_URL`, `env.ADMIN_TOKEN` to a location where they will only be written once after the tests start executing, else we will have duplicate entries.

Reply:
We wanted 1 jenkins job for 1 cluster provisioning use case since we want reporting to happen at this level.
So our assumption would be that there will be only 1 cluster definition in this file that needs to be looked at any point of time.